src/accelerate/hooks.py

```diff
@@ -155,17 +155,17 @@ def add_hook_to_module(module: nn.Module, hook: ModelHook, append: bool = False)
     module = hook.init_hook(module)
     module._hf_hook = hook
 
-    @functools.wraps(old_forward)
-    def new_forward(*args, **kwargs):
-        args, kwargs = module._hf_hook.pre_forward(module, *args, **kwargs)
-        if module._hf_hook.no_grad:
+    def new_forward(self, *args, **kwargs):
```

Comment:
Maybe replace `self` by `module` to not use the syntax of a class.

Reply:
Updated!